plutus-playground-client/default.nix

```diff
@@ -2,67 +2,80 @@
-      rm $out/.tidyoperators
-    '';
+  ghcWithPlutus = haskell.project.ghcWithPackages (ps: [ ps.plutus-core ps.plutus-tx ps.plutus-contract ps.plutus-ledger ps.playground-common ]);
 
   # generate-purescript: script to create purescript bridge code
   #
-  # * Note-1: We need to add ghc to the path because the purescript generator
+  # Note: We need to add ghc to the path because the purescript generator
   # actually invokes ghc to generate test data so we need ghc with the necessary deps
-  #
-  # * Note-2: This command is supposed to be available in the nix-shell but want
-  # to avoid plutus-core in the shell closure so we do $(nix-build ..) instead
   generate-purescript = pkgs.writeShellScriptBin "plutus-playground-generate-purs" ''
-    GHC_WITH_PKGS=${build-ghc-with-plutus}
+    if [ ! -d plutus-playground-client ]; then 
+      echo Please run plutus-playground-generate-purs from the root of the repository
+      exit 1
+    fi
+
+    GHC_WITH_PKGS=${ghcWithPlutus}
     export PATH=$GHC_WITH_PKGS/bin:$PATH
 
-    rm -rf ./generated
-    ${build-playground-exe}/bin/plutus-playground-server psgenerator generated
-    cd ..
-    echo Formatting files...
-    ${purs-tidy}/bin/purs-tidy format-in-place ./plutus-playground-client/generated
-    echo Done: formatted
+    generatedDir=./plutus-playground-client/generated
+    rm -rf $generatedDir
+
+    echo Generating purescript files in $generatedDir
+    ${playground-exe}/bin/plutus-playground-server psgenerator $generatedDir
+    echo -e Done generating purescript files
+
+    echo Formatting purescript files in $generatedDir
+    ${purs-tidy}/bin/purs-tidy format-in-place $generatedDir
+    echo Done formatting purescript files
   '';
 
   # start-backend: script to start the plutus-playground-server
   #
-  # Note-1: We need to add ghc to the path because the server provides /runghc
+  # Note: We need to add ghc to the path because the server provides /runghc
   # which needs ghc and dependencies.
-  # Note-2: We want to avoid to pull the huge closure in so we use $(nix-build) instead
   start-backend = pkgs.writeShellScriptBin "plutus-playground-server" ''
+    if [ ! -d plutus-playground-client ]; then 
+      echo Please run plutus-playground-server from the root of the repository
+      exit 1
+    fi
+
+    generatedDir=./plutus-playground-client/generated
+    if [ ! -d $generatedDir ]; then 
+      echo $generatedDir not found
+      plutus-playground-generate-purs
+    elif [ $# == 0 ]; then 
+      dirAge=$(datediff now $(date -r $generatedDir +%F))
+      echo Using Purescript files in $generatedDir which are $dirAge days old. 
+      echo Run plutus-playground-generate-purs -g to regenerate
+    elif [ "$1" == "-g" ]; then 
+       plutus-playground-generate-purs
+    fi 
+
+    echo
     echo "plutus-playground-server: for development use only"
-    GHC_WITH_PKGS=${build-ghc-with-plutus}
+    GHC_WITH_PKGS=${ghcWithPlutus}
     export PATH=$GHC_WITH_PKGS/bin:$PATH
```

Comment:
Again there is a tradeoff to be made here:

    Include plutus-core in the nix-shell once, but get plutus-playground-server to execute fast, or
    Don't include plutus-core in the nix-shell, saving some time, but on-demand every time plutus-playground-server (and plutus-playground-generate-purs) are called.